### PR TITLE
feat(data): implement GET /reading/resourcename/{resourcename} V2 API

### LINKED ISSUE #2967: [V2 data] Implement GET /reading/resourcename/{resourcename} V2 API
# 🚀 Feature Request

### Relevant Package

 This feature request is for core data

### Description

Per https://app.swaggerhub.com/apis-docs/EdgeXFoundry1/core-data/2.x#/default/get_reading_resourceName__resourceName_, implement GET /reading/resourcename/{resourcename} V2 API.

### PR DESCRIPTION
**Pull request by judehung**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-go/blob/master/.github/Contributing.md.

## Issue Number:
fix #2967 

## What is the new behavior?
Per https://app.swaggerhub.com/apis-docs/EdgeXFoundry1/core-data/2.x#/default/get_reading_resourceName__resourceName_, implement GET /reading/resourcename/{resourcename} V2 API.

Also update swagger yaml file to add 416 response when query range out of bound.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## New Imports

- [ ] Yes
- [x] No

### CODE REVIEW

File: internal/core/data/v2/application/reading_test.go [hunk 1]
```diff
@@ -55,6 +55,48 @@ func TestAllReadings(t *testing.T) {
...
+		offset             int
+		limit              int
+		resourceName       string
+		errorExpected      bool
+		ExpectedErrKind    errors.ErrKind
+		expectedCount      int
+		expectedStatusCode int
+	}{
+		{"Valid - all readings", 0, 20, testDeviceResourceName, false, "", len(readings), http.StatusOK},
+		{"Invalid - bounds out of range", len(readings) + 1, 10, testDeviceResourceName, true, errors.KindRangeNotSatisfiable, 0, http.StatusRequestedRangeNotSatisfiable},
```
Comment (lenny-intel) [inline internal/core/data/v2/application/reading_test.go:82]: Add test for missing resource name.

**Review by lenny-intel — CHANGES_REQUESTED** [review]

File: internal/core/data/v2/application/reading_test.go (hunk 1 above)
Comment (judehung) [reply internal/core/data/v2/application/reading_test.go:82]: Added.

**Comment by judehung** [comment]
Rebased and add unit test to check empty resource name per review comment.

**Review by lenny-intel — DISMISSED** [review]
LGTM

**Comment by judehung** [comment]
rebased

File: openapi/v2/core-data.yaml [hunk 2]
```diff
@@ -1346,6 +1346,18 @@ paths:
...
+          description: "Request range is not satisfiable"
+          headers:
+            X-Correlation-ID:
+              $ref: '#/components/headers/correlatedResponseHeader'
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+              examples:
+                400Example:
```
Comment (cloudxxx8) [inline openapi/v2/core-data.yaml:1359]: why not calling it 416Example?

Comment (judehung) [reply openapi/v2/core-data.yaml]: It's a typo since 416Example was added.  Fixed all occurrences of this typo.

**Review by cloudxxx8 — APPROVED** [review]
LGTM
